server/migrations.js

```diff
@@ -1,26 +1,28 @@
-Migrations.add({
-  version: 1,
-  name: 'Move rounds.progress data to new roundProgresses collection.',
-  up: function() {
-    var allRounds = Rounds.find({}).fetch();
-    allRounds.forEach(function(round) {
-      var done = 0;
-      var total = 0;
-      if(round.progress) {
-        done = round.progress.done;
-        total = round.progress.total;
-      }
-      RoundProgresses.insert({
-        roundId: round._id,
-        competitionId: round.competitionId,
-        done: done,
-        total: total,
+Migrations.add(
+  {
+    version: 2,
+    name: 'Compute rounds.totalCount and remove rounds.roundCode.',
+    up: function() {
+      var counts = {};
+
+      Rounds.find({}).fetch().forEach(function(round) {
+        var key = round.competitionId + ',' + round.eventCode;
+        counts[key] = (counts[key] || 0) + 1;
+      });
+
+      Object.keys(counts).forEach(function(key) {
+        ids = key.split(',');
+        Rounds.update(
+          {competitionId: ids[0], eventCode: ids[1]},
+          {$set: {totalRounds: counts[key]}},
+          {multi: true}
+        );
       });
 
-      Rounds.update({_id: round._id}, { $unset: { progress: 1 } });
-    });
+      Rounds.update({}, {$unset: { roundCode: 1 }}, {multi: true});
```

Comment:
I'm pretty sure this has nothing to do with what's in the database. See:

```bash
~/tmp/ccm @kaladin> meteor shell
> Rounds.update({}, {$unset: { totallyRandomAttribute: 1 }}, {multi: true});
Error: totallyRandomAttribute is not allowed by the schema
    at getErrorObject (packages/aldeed:collection2/collection2.js:369:1)
    at [object Object].doValidate (packages/aldeed:collection2/collection2.js:352:1)
    at [object Object].Mongo.Collection.(anonymous function) [as update] (packages/aldeed:collection2/collection2.js:154:1)
    at repl:1:8
    at /home/jeremy/tmp/ccm/.meteor/local/build/programs/server/shell-server.js:243:23
>
```

So, this migration won't run successfully on any fresh clones of ccm. I imagine you'll see this issue if you run a `meteor reset` and then `meteor`.